evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -74,7 +71,8 @@ def transform(self, X, y=None):
         X_ww = infer_feature_types(X)
         X_ww = X_ww.ww.rename({col: str(col) for col in X_ww.columns})
         es = self._make_entity_set(X_ww)
+
         feature_matrix = calculate_feature_matrix(features=self.features, entityset=es)
-        return _retain_custom_types_and_initalize_woodwork(
-            X_ww.ww.logical_types, feature_matrix
-        )
+        typed_columns = set(X_ww.columns).intersection(set(feature_matrix.columns))
```

Comment:
Do we need this set intersection? It seem it should be ok with just `feature_matrix.columns`?

Reply:
We need the set, since `calculate_feature_matrix` generates new columns that didn't exist in `X_ww`, whose schema these columns are passed into.